common/move_only.h

```diff
@@ -0,0 +1,22 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_MOVE_ONLY_H_
+#define CARBON_COMMON_MOVE_ONLY_H_
+
+namespace Carbon {
+
+// A base class that indicates a type is move-only. Typically this can be
+// achieved by declaring the move constructor and move assignment yourself; this
```

Comment:
I actually kind of prefer this approach if we could do it generally, just because it's a bit briefer and more explicit. i.e., instead of needing the two move constructors plus maybe a comment explaining what's going on, inherit from `MoveOnly`. But I assume you're more familiar with potential consequences of that than I am. :)

Reply:
I somewhat share that preference, but [the style guide says the usual way to achieve this is by explicitly declaring / deleting the operators in the class body](https://google.github.io/styleguide/cppguide.html#:~:text=This%20should%20usually%20take%20the%20form%20of%20explicitly%20declaring%20and/or%20deleting%20the%20appropriate%20operations%20in%20the%20public%20section%20of%20the%20declaration.), so I'm only suggesting to use a different technique here in the "non-usual" case where the style guide's advice is inapplicable.

Reply:
Let's start here, I feel like I should ask about a style change separately.